x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java

```diff
@@ -147,7 +149,23 @@ protected void doExecute(Task task, TermsEnumRequest request, ActionListener<Ter
         if (ccsCheckCompatibility) {
             checkCCSVersionCompatibility(request);
         }
-        new AsyncBroadcastAction(task, request, listener).start();
+        // log any errors that occur in a successful partial results scenario
+        ActionListener<TermsEnumResponse> loggingListener = listener.delegateFailureAndWrap((l, termsEnumResponse) -> {
+            // Deduplicate failures by exception message and index
+            ShardOperationFailedException[] deduplicated = ExceptionsHelper.groupBy(termsEnumResponse.getShardFailures());
```

Comment:
nit: just saw on debugging that calling groupBy even with an empty failures array creates a list and a HashSet, maybe we can check this either here or even inside the "groupBy" helper and shortcut for the majority case where we don't have any failures.

Reply:
Good catch. I'll change the helper to have an if-check guard.